module/CMakeLists.txt

```diff
@@ -323,25 +327,25 @@ if(DEAL_II_WITH_CXX20_MODULE)
       BASE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/../include/
       )
 
+    #
+    # In addition, link in all objects from the bundled object targets
+    # (listed in the DEAL_II_BUNDLED_TARGETS_* global property).
+    #
+    get_property(_bundled_object_targets GLOBAL PROPERTY DEAL_II_BUNDLED_TARGETS_${build})
+    set(_object_files "")
+    foreach(_target ${_bundled_object_targets})
+      list(APPEND _object_files "$<TARGET_OBJECTS:${_target}>")
+    endforeach()
+    target_sources(${DEAL_II_TARGET_NAME}_module_${build_lowercase}
+      PRIVATE ${_object_files}
+      )
+
     # Ensure that all .inst files and the macro headers are in place
     add_dependencies(${DEAL_II_TARGET_NAME}_module_${build_lowercase}
       expand_all_instantiations
       build_macros_header
       )
 
-    # There are other sources that go into our libraries, namely the
-    # files in the bundled/ directories that we compile. We could add
-    # those to the list of sources, but perhaps simpler is to just
-    # link in the object library we create from them to link into the
-    # non-module library. This implies that they are compiled with the
-    # non-module compile flags, and consequently mangled as non-module
-    # symbols, but that's ok -- these are all symbols that are not
-    # part of the module anyway.
-    get_property(_bundled_object_targets GLOBAL PROPERTY DEAL_II_BUNDLED_TARGETS_${build})
-    target_link_libraries(${DEAL_II_TARGET_NAME}_module_${build_lowercase}
-      PRIVATE ${_bundled_object_targets}
-      )
-
```

Comment:
Thanks for moving the change back to the right place. (And yes, the right place is above the dependencies where you put it -- thank you!)